src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -82,14 +81,15 @@ export interface IQuickPickTerminalObject {
 	keyMods: IKeyMods | undefined;
 }
 
-export interface ICommandTracker {
-	scrollToPreviousCommand(): void;
-	scrollToNextCommand(): void;
-	selectToPreviousCommand(): void;
-	selectToNextCommand(): void;
+export interface IMarkTracker {
+	scrollToPreviousMark(): void;
+	scrollToNextMark(): void;
+	selectToPreviousMark(): void;
+	selectToNextMark(): void;
 	selectToPreviousLine(): void;
 	selectToNextLine(): void;
 	clearMarker(): void;
+	scrollToClosestMark(startMarker: IMarker, endMarker?: IMarker, highlight?: boolean): void;
```

Comment:
I think `Marker` is more appropriate here as commands are markers, marks are markers, but commands are not marks. I suggest whenever the string ID is important we talk about a "mark", and when we're just working with an `IMarker` we use "marker".